src/mlpack/core/util/split_data.hpp

```diff
@@ -0,0 +1,155 @@
+#ifndef __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+#define __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+
+#include <mlpack/core.hpp>
+
+#include <iterator>
+#include <numeric>
+#include <random>
+#include <tuple>
+#include <vector>
+
+namespace mlpack {
+namespace util {
+
+/**
+ *Split training data and test data, please define
+ *ARMA_USE_CXX11 to enable move of c++11
+ */
+class TrainTestSplit
+{
+public:
+  /**
+   * @brief TrainTestSplit
+   * @param testRatio the ratio of test data
+   * @param seed seed of the random device
+   * @warning slice should not less than 1
+   */
+  TrainTestSplit(double testRatio,                
+                 arma::arma_rng::seed_type seed = 0) :
+    seed(seed),    
+    testRatio(testRatio)
+  {    
+  }
+
+  /**
+   *Split training data and test data, please define
+   *ARMA_USE_CXX11 to enable move of c++11
```

Comment:
mlpack requires C++11 so I think we can assume that `ARMA_USE_CXX11` is enabled without any issues.